### PR TITLE
Rotate promos

### PR DESCRIPTION
**Pull request by james-prado**
Closes https://github.com/MyEtherWallet/MyEtherWallet/issues/750

- Seperate promos into individual components
	- creating banners is becoming too difficult with a one-size-fits-all component, separating them makes them easier to write and compose
- Rotate promos every 10 seconds
- Add fade animation when transitioning between promos

### CODE REVIEW

File: common/components/BalanceSidebar/Promos.scss [hunk 1]
```diff
@@ -88,3 +107,13 @@
...
   }
 }
+
+.carousel-exit {
+  opacity: 1;
+  transform: translate(0%, -100%);
+  transition: opacity 300ms;
+  pointer-events: none;
+  &.carousel-exit-active {
+    opacity: 0.01;
```
Comment (wbobeirne) [inline common/components/BalanceSidebar/Promos.scss]: You can just make it zero no problem.

File: common/components/BalanceSidebar/Promos.scss [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
   &-promo {
     position: relative;
-    height: 6rem;
+    height: inherit;
     display: block;
     color: #fff;
     text-decoration: none;
     text-align: center;
-    transition-duration: 200ms;
+    transition: opacity 200ms;
```
Comment (wbobeirne) [inline common/components/BalanceSidebar/Promos.scss]: You can just kill this if it's being handled in `.carousel-exit`

File: common/components/BalanceSidebar/Promos.tsx [hunk 3]
```diff
@@ -94,6 +58,15 @@ export default class Promos extends React.Component<{}, State> {
   }
 
   private navigateToPromo = (idx: number) => () => {
+    // stop rotating when user begins interacting with promos
+    clearInterval(this.timer);
```
Comment (wbobeirne) [inline common/components/BalanceSidebar/Promos.tsx]: This is a nice touch I was going to suggest, glad you beat me to the punch!

File: common/components/BalanceSidebar/Promos.tsx [hunk 4]
```diff
@@ -94,6 +58,15 @@ export default class Promos extends React.Component<{}, State> {
...
     this.setState({ activePromo: Math.max(0, Math.min(promos.length, idx)) });
   };
+
+  private rotate = () => {
+    const previousIndex = this.state.activePromo;
+    const nextIndex = previousIndex + 1;
+    const nextIndexExists = typeof promos[nextIndex] !== 'undefined' ? true : false;
```
Comment (wbobeirne) [inline common/components/BalanceSidebar/Promos.tsx]: You can make this code _much_ simpler with modular math:
```
const activePromo = (this.state.activePromo + 1) % promos.length;
this.setState({ activePromo });
```
The % sign is a modulo operator, which causes numbers to wrap.

File: common/components/BalanceSidebar/Promos.scss (hunk 2 above)
Comment (james-prado) [reply common/components/BalanceSidebar/Promos.scss]: the transition here is for `:hover` effect

File: common/components/BalanceSidebar/Promos.tsx (hunk 3 above)
Comment (james-prado) [reply common/components/BalanceSidebar/Promos.tsx]: Thx 😄

File: common/components/BalanceSidebar/Promos.tsx (hunk 4 above)
Comment (james-prado) [reply common/components/BalanceSidebar/Promos.tsx]: Good use of modulo 👍 , thx for the suggestion

**Pull request merged**